src/main/scala/edu/berkeley/ce/rockslicing/Block.scala

```diff
@@ -335,8 +392,9 @@ case class Block(center: (Double,Double,Double), faces: Seq[Face]) {
           Delaunay.Vector2(rotatedVertex(0), rotatedVertex(1))
         }.distinct.toList
 
-        // If z-component of normal is negative, order needs to be reversed to maintain clockwise orientation
-        if (face.c < -NumericUtils.EPSILON) {
+        // If normal is -z-axis, order needs to be reversed to maintain clockwise orientation
+        // rotation matrix is identity matrix in this case
+        if (math.abs(face.c + 1.0) < NumericUtils.EPSILON) {
```

Comment:
The rotation matrix actually takes care of all cases except when the normal vectors are parallel, so it's not necessary to do anything unless the normal vectors are parallel but point in opposite directions.